src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -323,33 +340,11 @@ private void processJibriIqFromMeet(JibriIq iq, XmppChatMember sender)
             if (jibriJid == null)
             {
                 sendErrorResponse(
-                    iq, XMPPError.Condition.service_unavailable, null);
+                        iq, XMPPError.Condition.service_unavailable, null);
                 return;
             }
 
-            final String roomName = MucUtil.extractName(senderMucJid);
-
-            try
-            {
-                final XMPPError err = startJibri(
-                        jibriJid, roomName, iq.getStreamId());
-                if (err == null)
-                {
-                    // ACK the original request
-                    sendResultResponse(iq);
-                }
-                else
-                {
-                    sendErrorResponse(
-                            iq, XMPPError.Condition.interna_server_error, null);
-                }
-            }
-            catch (final SmackException.NoResponseException e)
-            {
-                sendErrorResponse(
-                        iq, XMPPError.Condition.request_timeout, null);
-                return;
-            }
+            scheduleRetryStartJibri(jibriJid, iq);
```

Comment:
I was just trying to think of a way to ensure resources get cleaned up at the end of a function (regardless of how or where the function returns) since that was the only reason I could think of for returns being bad. I suppose that doing it this way wouldn't work because of how Java handles exceptions though. I just discovered Java 7's "try with resource"; I suppose that's the closest Java thing.